datalad/cmd.py

```diff
@@ -156,54 +208,67 @@ def __call__(self, cmds):
         str or list
           Output received from process.  list in case if cmds was a list
         """
-        input_multiple = isinstance(cmds, list)
-        if not input_multiple:
-            cmds = [cmds]
+        self._active += 1
+        try:
+            input_multiple = isinstance(cmds, list)
+            if not input_multiple:
+                cmds = [cmds]
 
-        output = [o for o in self.yield_(cmds)]
-        return output if input_multiple else output[0]
+            output = [o for o in self.yield_(cmds)]
+            return output if input_multiple else output[0]
+        finally:
+            self._active -= 1
```

Comment:
A method decorator would require special handling for `yield_()`, since it's a generator.  In fact, I'm pretty sure we would have to write two separate decorators, one for `yield_()` and one for `__call__()` and `proc1()`, and at that point there'd barely be any benefit.

Reply:
There could be a decorator which handles both.  We have one like that here https://github.com/datalad/datalad/blob/HEAD/datalad/interface/utils.py#L467 I believe.

But indeed might be a bit too much for such a limited single use case. So I am ok if other do not object as well